### ISSUE TITLE
Fix false positives for escaped colons in selector-pseudo-element-colon-notation

### ISSUE THREAD

**Issue by hacknug** [stylelint/stylelint#3310]
**Describe the bug? (A clear and concise description of what the bug is)**

Stylelint expects a class named `.pseudo\:before-none` to have two colons even if the colon in the name is escaped and is not targeting any pseudo elements.

I'm using this _weird_ naming convention to make my classes match Tailwind's.

**Which rule, if any, is this issue related to?**

`selector-pseudo-element-colon-notation`

**What CSS is needed to reproduce this issue?**

```css
.pseudo\:before-none::before { content: none; }
.pseudo\:after-none::after { content: none; }
```

**What stylelint configuration is needed to reproduce this issue?**

```json
{
  "rules": {
    "selector-pseudo-element-colon-notation": "double",
  }
}
```

**Which version of stylelint are you using?**

`9.2.0`

**How are you running stylelint: CLI, PostCSS plugin, Node API?**

Via [vscode-stylelint](https://github.com/shinnn/vscode-stylelint)

**Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?**

No.

**What did you expect to happen?**

No warnings to be flagged when the colon in `:before` or `:after` is escaped.

**What actually happened (e.g. what warnings or errors you are getting)?**

The following warnings were flagged:

```shell
Expected double colon pseudo-element notation (selector-pseudo-element-colon-notation)
```

**Comment by chriseppstein** [comment]
I’m pretty sure this is fixed with the upcoming release https://github.com/postcss/postcss-selector-parser/blob/master/src/__tests__/escapes.js#L9

**Comment by chriseppstein** [comment]
@jeddy3 No need for documentation from @hacknug. The escape should prevent it from being seen by the browser as a pseudo.

**Comment by vankop** [comment]
@evilebottnawi works in latest postcss-selector-parser version https://github.com/postcss/postcss-selector-parser/pull/207
